packages/wallets/ONTO/src/adapter.ts

```diff
@@ -0,0 +1,162 @@
+import {
+    Adapter,
+    BaseSignerWalletAdapter,
+    scopePollingDetectionStrategy,
+    WalletAccountError,
+    WalletName,
+    WalletNotConnectedError,
+    WalletNotReadyError,
+    WalletPublicKeyError,
+    WalletReadyState,
+    WalletSignTransactionError,
+} from '@solana/wallet-adapter-base';
+import { PublicKey, Transaction } from '@solana/web3.js';
+import bs58 from 'bs58';
+
+interface OntoWallet {
+    isOnto?: boolean;
+    signTransaction(transaction: Transaction): Promise<Transaction>;
+    isConnected(): boolean;
+    connect(): Promise<string[]>;
+    disconnect(): Promise<void>;
+    request(params: { method: string; params: string | string[] | unknown }): Promise<{
+        signature: string;
+        publicKey: string;
+    }>;
+}
+
+interface OntoWindow extends Window {
+    onto?: {
+        sol?: OntoWallet;
+    };
+}
+
+declare const window: OntoWindow;
+
+export interface OntoWalletAdapterConfig {}
+
+export const OntoWalletName = 'ONTO' as WalletName;
+
+export class Coin98WalletAdapter extends BaseSignerWalletAdapter {
```

Comment:
This was copied from Coin98, please fix and review for any other copied sections.